### PR TITLE
[sdk/python] Fix secret regression

### LINKED ISSUE #5492: Regression around AdditionSecretOutputs in Python
The following code should mark "result" as a secret but does not

```
import pulumi
import pulumi_random
password = pulumi_random.RandomPassword("super-secret-password", length=20, special=True,
                                        opts=pulumi.ResourceOptions(additional_secret_outputs=['result']))
pulumi.export("secret", password.result)
```

The output looks as follows:

```
pulumi up
Previewing update (dev)
View Live: https://app.pulumi.com/stack72/test-py-secrets/dev/previews/679cd80d-f7ea-4d82-b00c-fd6655df18c6
     Type                            Name                   Plan
 +   pulumi:pulumi:Stack             test-py-secrets-dev    create
 +   └─ random:index:RandomPassword  super-secret-password  create
Resources:
    + 2 to create
Do you want to perform this update? yes
Updating (dev)
View Live: https://app.pulumi.com/stack72/test/dev/updates/1
     Type                            Name                   Status
 +   pulumi:pulumi:Stack             test-py-secrets-dev    created
 +   └─ random:index:RandomPassword  super-secret-password  created
Outputs:
    secret: "(VNcUMh=U+DIT!vlosU8"
Resources:
    + 2 created
```

The equivalent code currently works in NodeJs

```
pulumi up
Previewing update (dev)
View Live: https://app.pulumi.com/stack72/test-nodejs-secrets/dev/previews/86a06366-4a40-4eee-9720-08f6196ae412
     Type                            Name                     Plan
 +   pulumi:pulumi:Stack             test-nodejs-secrets-dev  create
 +   └─ random:index:RandomPassword  test                     create
Resources:
    + 2 to create
Do you want to perform this update? yes
Updating (dev)
View Live: https://app.pulumi.com/stack72/test-nodejs-secrets/dev/updates/1
     Type                            Name                     Status
 +   pulumi:pulumi:Stack             test-nodejs-secrets-dev  created
 +   └─ random:index:RandomPassword  test                     created
Outputs:
    y: "[secret]"
Resources:
    + 2 created
Duration: 4s
```

pulumi v2.11.1

Not sure when this started happening

### PR DESCRIPTION
**Pull request by justinvp**
A recent change to output deserialization resulted in secrets being returned unwrapped. This change addresses the regression, ensuring any unwrapped secret values are rewrapped before being returned.

Fixes #5492

### CODE REVIEW

File: sdk/python/lib/pulumi/runtime/rpc.py [hunk 1]
```diff
@@ -418,9 +421,12 @@ def translate_output_properties(output: Any,
     :param Optional[type] typ: The output's target type.
     """
 
-    # If it's a secret, unwrap the value so the output is in alignment with the expected type.
+    # If it's a secret, unwrap the value so the output is in alignment with the expected type, call
+    # translate_output_properties with the unwrapped value, and then rewrap the result as a secret.
     if is_rpc_secret(output):
-        output = unwrap_rpc_secret(output)
```
Comment (justinvp) [inline sdk/python/lib/pulumi/runtime/rpc.py:423]: This was the problem 🤦😢

File: tests/integration/integration_nodejs_test.go [hunk 2]
```diff
@@ -489,6 +489,44 @@ func TestProviderSecretConfig(t *testing.T) {
 	})
 }
 
+func TestResourceWithSecretSerializationNodejs(t *testing.T) {
```
Comment (justinvp) [inline tests/integration/integration_nodejs_test.go:492]: We had this existing test in `tests/integration/secret_outputs` from https://github.com/pulumi/pulumi/pull/2875 but we somehow lost the actual Go code to run the test at some point. I added it back for Node.js, and augmented it with use of `additionalSecretOutputs`, and also added the same test for Python.

**Review by leezen — APPROVED** [review]

File: tests/integration/integration_nodejs_test.go [hunk 3]
```diff
@@ -489,6 +489,44 @@ func TestProviderSecretConfig(t *testing.T) {
...
+		Dir:          filepath.Join("secret_outputs", "nodejs"),
+		Dependencies: []string{"@pulumi/pulumi"},
+		Quick:        true,
+		ExtraRuntimeValidation: func(t *testing.T, stackInfo integration.RuntimeValidationStackInfo) {
+			// The program exports three resources:
+			//   1. One named `withSecret` who's prefix property should be secret, specified via `pulumi.secret()`.
+			//   2. One named `withSecretAdditional` who's prefix property should be a secret, specified via
+			//      additionalSecretOutputs.
+			//   3. One named `withoutSecret` which should not be a secret.
+			// We serialize both of the these as POJO objects, so they appear as maps in the output.
```
Comment (komalali) [inline tests/integration/integration_nodejs_test.go:503]: What's a POJO?

**Review by komalali — APPROVED** [review]

File: tests/integration/integration_nodejs_test.go (hunk 3 above)
Comment (justinvp) [reply tests/integration/integration_nodejs_test.go:503]: I was tying to find a good definition. https://masteringjs.io/tutorials/fundamentals/pojo is prob a good summary.

Comment (komalali) [reply tests/integration/integration_nodejs_test.go:503]: aha thank you!

**Pull request merged**